packages/web-components/src/design-system-provider/index.ts

```diff
@@ -92,7 +95,13 @@ function designToken<T>(token: DesignToken<T>) {
   return (source: DesignSystemProvider, key: string) => {
     source[key + 'Changed'] = function (this: DesignSystemProvider, prev: T | undefined, next: T | undefined) {
       if (next !== undefined && next !== null) {
-        token.setValueFor(this, next as DesignTokenValue<T>);
+        if (key === "fillColor") {
```

Comment:
Down below you've declared the fillColor property to be a swatch but here you're asserting it's a string, so I think this is a mis-match

Reply:
This code is probably cheating. All other design system attributes here were simple types, so I wasn't sure if this is correct, but at least the way it is working now allows for a hex color to be provided in the html element attribute, but to parse it and set the design token to the correct type. Before I added this, the design token was actually getting set to a string and the recipes were failing because they expected a Swatch. Looking at it again, is the `converter` the right thing to do here? I can try that.

Reply:
All *attributes* are simple (stringable) types, but you could store the value as a property.

If you want to maintain it as an attribute, then you *could* use a converter but the API would be a little odd, because the property would be typed as a `Swatch` but you apply a hex string via an attribute.